keps/sig-api-machinery/5073-declarative-validation-with-validation-gen/README.md

```diff
@@ -1024,10 +1028,49 @@ The streamed data does not require declarative validation, as it is not structur
+| Slice values<br/>`listType=map`      | -                                                              | (re)validate items which are not found (by key) in the old slice or are changed | same                                                                                                                |
+| Slice values<br/>`listType=set`      | -                                                              | validate items which are not found (by value) in the old slice                 | Validate all elements (CRDs ratcheting may be expanded to match in-tree ratcheting)                                  |
+| Maps                                | all elements are equivalent                                    | revalidate the map if any element changed                                      | same                                                                                                                |
+| Map values<br/>`mapType=granular`    | -                                                              | (re)validate items which are not found (by key) in the old map                 | same                                                                                                                |
+| Map values<br/>`mapType=atomic`      | -                                                              | (re)validate items which are not found (by key) in the old map                 | [Issue #131566](https://github.com/kubernetes/kubernetes/issues/131566) (Alignment needed)                     |
+
+**Note on Atomic Types:** The behavior for `structType=atomic` and `mapType=atomic` intentionally deviates from strict atomic re-validation. Only the specific sub-fields or key-value pairs *that were actually modified* are re-validated. This prioritizes user experience but requires alignment with CRD behavior (tracked in Issue #131566).
+
+#### Ratcheting and Cross-Field Validation
+
+A challenge arises if a cross-field validation rule (e.g. `X < Y`) is defined on a common ancestor struct, and an unrelated field (e.g. `Z`) within that same ancestor is modified. This change to `Z` makes the common ancestor “changed” overall, triggering re-validation of the `X < Y` rule. If this rule was recently evolved (e.g., made stricter), it might now fail even if `X` and `Y` themselves are not modified by the user’s update. This could violate the principle “Unchanged fields do not cause update rejections.”
+
+For the initial implementation, this behavior will be documented. Furthermore, cross-field validation rules themselves must incorporate internal ratcheting logic. For instance, generated code for dedicated cross-field tags (like `+k8s:unionMember`) will be designed to only act upon changes to the specific fields they govern and were designed to validate.
```

Comment:
Please build us an example as soon as we have enough merged so we can see how painful (or not) it seems to be.